conjure-java-jackson-serialization/src/main/java/com/palantir/conjure/java/serialization/InstrumentedJsonFactory.java

```diff
@@ -0,0 +1,710 @@
+            return delegate.getTokenLocation();
+        }
+
+        @Override
+        public Object currentValue() {
+            return delegate.currentValue();
+        }
+
+        @Override
+        public void assignCurrentValue(Object value) {
+            delegate.assignCurrentValue(value);
+        }
+
+        @Override
+        public Object getCurrentValue() {
+            return delegate.getCurrentValue();
+        }
+
+        @Override
+        public void setCurrentValue(Object value) {
+            delegate.setCurrentValue(value);
+        }
+
+        @Override
+        public int releaseBuffered(OutputStream out) throws IOException {
+            return delegate.releaseBuffered(out);
+        }
+
+        @Override
+        public int releaseBuffered(Writer writer) throws IOException {
+            return delegate.releaseBuffered(writer);
+        }
+
+        @Override
+        public JsonParser enable(Feature feature) {
+            delegate.enable(feature);
+            return this;
+        }
+
+        @Override
+        public JsonParser disable(Feature feature) {
+            delegate.disable(feature);
+            return this;
+        }
+
+        @Override
+        public JsonParser configure(Feature feature, boolean state) {
+            delegate.configure(feature, state);
+            return this;
+        }
+
+        @Override
+        public boolean isEnabled(Feature feature) {
+            return delegate.isEnabled(feature);
+        }
+
+        @Override
+        public boolean isEnabled(StreamReadFeature feature) {
+            return delegate.isEnabled(feature);
+        }
+
+        @Override
+        public int getFeatureMask() {
+            return delegate.getFeatureMask();
+        }
+
+        @Override
+        @Deprecated
+        public JsonParser setFeatureMask(int mask) {
+            delegate.setFeatureMask(mask);
+            return this;
+        }
+
+        @Override
+        public JsonParser overrideStdFeatures(int values, int mask) {
+            delegate.overrideStdFeatures(values, mask);
+            return this;
+        }
+
+        @Override
+        public int getFormatFeatures() {
+            return delegate.getFormatFeatures();
+        }
+
+        @Override
+        public JsonParser overrideFormatFeatures(int values, int mask) {
+            delegate.overrideFormatFeatures(values, mask);
+            return this;
+        }
+
+        @Override
+        public JsonToken nextToken() throws IOException {
+            return delegate.nextToken();
+        }
+
+        @Override
+        public JsonToken nextValue() throws IOException {
+            return delegate.nextValue();
+        }
+
+        @Override
+        public boolean nextFieldName(SerializableString str) throws IOException {
+            return delegate.nextFieldName(str);
+        }
+
+        @Override
+        public String nextFieldName() throws IOException {
+            return delegate.nextFieldName();
```

Comment:
Do we want to track field name lengths as well?

[Looking at `StreamReadConstraints`](https://github.com/FasterXML/jackson-core/blob/2.15/src/main/java/com/fasterxml/jackson/core/StreamReadConstraints.java#L30-L43) we probably want to track number length (default limit `1000`), nesting depth (default limit `1000`), and string length (default limit `5_000_000`)

Reply:
Ideally I'd like to track number length, however it's tricky because precision may be lost when we read a value, and we may need to work backwards to understand the input data size.